src/jsonrpc-mgr.h

```diff
@@ -14,31 +14,34 @@
 // with this program (see COPYING). If not, see <https://www.gnu.org/licenses/>.
 //
 
-#ifndef JSONRPC_MG_MGR_H
-#define JSONRPC_MG_MGR_H
+#ifndef JSONRPC_MGR_H
+#define JSONRPC_MGR_H
 
-#include <boost/container/static_vector.hpp>
-#include <mongoose.h>
 #include <string>
 
-namespace cartesi {
+#include "asio-config.h" // must be included before any ASIO header
```

Comment:
This is producing warnings because of Boost's use of sprintf. This patch removes them.

```
diff --git a/src/jsonrpc-mgr.h b/src/jsonrpc-mgr.h
index dc7ee72..31de926 100644
--- a/src/jsonrpc-mgr.h
+++ b/src/jsonrpc-mgr.h
@@ -19,10 +19,13 @@
 
 #include <string>
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 #include "asio-config.h" // must be included before any ASIO header
 #include <boost/asio/io_context.hpp>
 #include <boost/beast/core/tcp_stream.hpp>
 #include <boost/container/static_vector.hpp>
+#pragma GCC diagnostic pop
 
 namespace cartesi {
 
diff --git a/src/jsonrpc-remote-machine.cpp b/src/jsonrpc-remote-machine.cpp
index 85cd13a..bc697b0 100644
--- a/src/jsonrpc-remote-machine.cpp
+++ b/src/jsonrpc-remote-machine.cpp
@@ -33,12 +33,15 @@
 #include <sys/wait.h>
 #include <unistd.h>
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 #include "asio-config.h" // must be included before any ASIO header
 #include <boost/asio/ip/tcp.hpp>
 #include <boost/asio/signal_set.hpp>
 #include <boost/beast/core.hpp>
 #include <boost/beast/http.hpp>
 #include <boost/beast/version.hpp>
+#pragma GCC diagnostic pop
 
 #include "base64.h"
 #include "json-util.h"
diff --git a/src/jsonrpc-virtual-machine.cpp b/src/jsonrpc-virtual-machine.cpp
index 79731d2..ef0d35c 100644
--- a/src/jsonrpc-virtual-machine.cpp
+++ b/src/jsonrpc-virtual-machine.cpp
@@ -21,12 +21,15 @@
 #include <cstdint>
 #include <string>
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 #include "asio-config.h" // must be included before any ASIO header
 #include <boost/asio/connect.hpp>
 #include <boost/asio/ip/tcp.hpp>
 #include <boost/beast/core.hpp>
 #include <boost/beast/http.hpp>
 #include <boost/beast/version.hpp>
+#pragma GCC diagnostic pop
 
 #include <boost/type_index.hpp>
 
```

Reply:
Applied this diff.